lib/dao.js

```diff
@@ -2334,6 +2335,12 @@ DataAccessObject.prototype.updateAttributes = function updateAttributes(data, op
     if (err) return cb(err);
     data = ctx.data;
 
+    if (strict) {
+      var keys = Object.keys(Model.definition.properties);
```

Comment:
`strict` also supports other setting such as `throw` or `validate`. Do we want to call back with such errors for those cases?